tests/ParserTest.php

```diff
@@ -48,14 +48,18 @@ public function testParserResetsTagsBetweenFeatures(): void
     {
         $parser = $this->createGherkinParser();
 
-        $parser->parse(
-            <<<'FEATURE'
-            Feature:
-            Scenario:
-            Given step
-            @skipped
-            FEATURE
-        );
+        try {
+            $parser->parse(
+                <<<'FEATURE'
+                Feature:
+                Scenario:
+                Given step
+                @skipped
+                FEATURE,
+            );
+        } catch (ParserException) {
+            // expected - features cannot end with tags
+        }
```

Comment:
This test relied on us parsing tags at the end of a file, but I don't think that was expected end-user behaviour. It was simply to prove that the parser state resets correctly on moving to the next file, and since we already know that tags are reset between features / scenarios / etc, the test had to create an artificial case where the tag was not attached to any node.